AppDB/backup/backup_recovery_helper.py

```diff
@@ -211,3 +226,152 @@ def untar_backup_files(source):
+    storage: A str, one of the StorageTypes class members.
+    bucket_name: A str, the name of the bucket to restore apps from.
+  Returns:
+    True on success, False otherwise.
+  """
+  # Download from GCS to backups location.
+  if storage == StorageTypes.GCS:
+    objects = gcs_helper.list_bucket(bucket_name)
+    for app_path in objects:
+      if not app_path.startswith(gcs_helper.APPS_GCS_PREFIX):
+        continue
+
+      # Only keep the relative name of the app file.
+      # E.g. myapp.tar.gz (app_file) out of apps/myapp.tar.gz (app_path)
+      app_file = app_path[len(gcs_helper.APPS_GCS_PREFIX):]
+      source = 'gs://{0}/{1}'.format(bucket_name, app_path)
+      destination = '{0}/{1}'.format(APP_BACKUP_DIR_LOCATION, app_file)
+      if not gcs_helper.download_from_bucket(source, destination):
+        logging.error("Error while downloading '{0}' from GCS.".format(source))
+        delete_app_tars(APP_BACKUP_DIR_LOCATION)
+        return False
+
+  # Copy source code tars from backups location to /opt/appscale/apps.
+  apps_to_deploy = []
+  for dir_path, _, filenames in os.walk(APP_BACKUP_DIR_LOCATION):
+    for filename in filenames:
+      source = '{0}/{1}'.format(dir_path, filename)
+      destination = '{0}/{1}'.format(APP_DIR_LOCATION, filename)
+      try:
+        shutil.copy(source, destination)
+        apps_to_deploy.append(destination)
+      except:
+        logging.error("Error while restoring '{0}'.".format(source))
+        if storage == StorageTypes.GCS:
+          delete_app_tars(APP_BACKUP_DIR_LOCATION)
+        delete_app_tars(APP_DIR_LOCATION)
+        return False
+
+  # Deploy apps.
+  if not deploy_apps(apps_to_deploy):
+    logging.error("Failed to successfully deploy one or more of the "
+      "following apps: {0}".format(apps_to_deploy))
+    if storage == StorageTypes.GCS:
+      delete_app_tars(APP_BACKUP_DIR_LOCATION)
+    delete_app_tars(APP_DIR_LOCATION)
+    return False
+
+  # Cleanup.
+  if storage == StorageTypes.GCS:
+    delete_app_tars(APP_BACKUP_DIR_LOCATION)
+
+  return True
+
+def delete_app_tars(location):
+  """ Deletes applications tars from the designated location.
+
+  Args:
+    location: A str, the path to the application tar(s) to be deleted.
+  Returns:
+    True on success, False otherwise.
+  """
+  for dir_path, _, filenames in os.walk(location):
+    for filename in filenames:
+      if not remove('{0}/{1}'.format(dir_path, filename)):
+        return False
+  return True
+
+def deploy_apps(app_paths):
+  """ Deploys all apps that reside in /opt/appscale/apps.
+
+  Args:
+    app_paths: A list of the full paths of the apps to be deployed.
+  Returns:
+    True on success, False otherwise.
+  """
+  uaserver = SOAPpy.SOAPProxy('https://{0}:{1}'.format(
+    appscale_info.get_db_master_ip(), UA_SERVER_PORT))
+
+  acc = AppControllerClient(appscale_info.get_login_ip(),
+    appscale_info.get_secret())
+
+  # Wait for Cassandra to come up after a restore.
+  time.sleep(5)
```

Comment:
Is there a need to check if Cassandra is up before continuing, or is 5 sec always safe? If so, should this be a constant?